### PR TITLE
Feature/apl 522 implement secure transport control

### PR DESCRIPTION
**Pull request by nemezxxx**

### CODE REVIEW

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/rest/service/TransportInteractionServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+public class TransportInteractionServiceImpl implements TransportInteractionService {
+    
+    
+    private static final Logger log = LoggerFactory.getLogger(TransportInteractionServiceImpl.class);
+    private String wsUrl; 
+
+    private TransportInteractionWebSocket transportInteractionWebSocket;
+    
+    @Setter
+    private boolean done;
```
Comment (AndrewBoyarsky) [inline apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/rest/service/TransportInteractionServiceImpl.java]: done is not thread safe. Use volatile keyword to guarantee synchronized read/write for this value.

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/rest/service/TransportInteractionServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+            transportInteractionWebSocket = new TransportInteractionWebSocket(new URI(wsUrl));            
+            Runnable task = () -> {
+                for(;;) {                    
+                    try {                        
+                        Thread.sleep(1000);
+                    } catch (InterruptedException ex) {
+                        log.error( "Runnable exception: {} ", ex.toString() );
+                    }                    
+                    transportInteractionWebSocket.tick();                   
+                    if (done) break;
```
Comment (AndrewBoyarsky) [inline apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/rest/service/TransportInteractionServiceImpl.java]: Looks like 'done' will be always 'false'

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/rest/service/TransportInteractionWebSocket.java [hunk 3]
```diff
@@ -0,0 +1,381 @@
...
+                case PENDING        : return "PENDING";
+                case CONNECTED      : return "CONNECTED";
+                case DISCONNECTED   : return "DISCONNECTED";
+            };
+        return null;
+        }                
+    }
+    
+    @Getter
+    String remoteip;
```
Comment (AndrewBoyarsky) [inline apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/rest/service/TransportInteractionWebSocket.java]: Maybe change all variable names to camel case?

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/rest/service/TransportInteractionWebSocket.java [hunk 4]
```diff
@@ -0,0 +1,381 @@
...
+        try {
+
+        secureTransportStatus = SecureTransportStatus.INITIAL;
+        cleanupComParams();
+
+        this.closeLatch = new CountDownLatch(1);
+        client = new WebSocketClient();
+        client.start();
+        ClientUpgradeRequest request = new ClientUpgradeRequest();
+        client.connect(this, endpointURI, request);
```
Comment (AndrewBoyarsky) [inline apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/rest/service/TransportInteractionWebSocket.java]: When connect fails, client will be not shutdown

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/rest/service/TransportInteractionWebSocket.java [hunk 5]
```diff
@@ -0,0 +1,381 @@
...
+     *
+     * @param message
+     */
+    public void sendMessage(String message) {        
+        try {
+            Future<Void> fut;            
+            fut = session.getRemote().sendStringByFuture(message);
+            fut.get(2, TimeUnit.SECONDS);
+        } catch (InterruptedException ex) {
+            log.error ("sendMessage InterruptedException: {}", ex.getMessage().toString() );
```
Comment (AndrewBoyarsky) [inline apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/rest/service/TransportInteractionWebSocket.java]: Looks like error handling is the same for all exceptions, combine it in one catch block

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/rest/service/TransportInteractionWebSocket.java [hunk 6]
```diff
@@ -0,0 +1,381 @@
...
+
+    /**
+     * Inquiring status of connection from transport service
+     */
+
+    private void getSecureTransportStatus( ) {                
+        log.debug("TransportInteractionWebSocket: getSecureTransportStatus");
+        TransportStopRequest stopRequest = new TransportStopRequest();
+        stopRequest.type = "GETSTATUSREQUEST";
+        Random rand = new Random();
```
Comment (AndrewBoyarsky) [inline apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/rest/service/TransportInteractionWebSocket.java]: Maybe move random to class field?

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/rest/service/TransportInteractionWebSocket.java [hunk 7]
```diff
@@ -0,0 +1,381 @@
...
+        stopRequest.id = rand.nextInt(255);          
+        ObjectMapper mapper = new ObjectMapper();
```
Comment (AndrewBoyarsky) [inline apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/rest/service/TransportInteractionWebSocket.java]: Maybe move mapper to  static class field?

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/app/AplCore.java [hunk 8]
```diff
@@ -168,6 +177,8 @@ private void startUp() {
                 aplAppStatus.durableTaskUpdate(initCoreTaskID,  5.0, "API initialization done");
 
 //                CDI.current().select(NtpTime.class).get().start();
+               aplAppStatus.durableTaskUpdate(initCoreTaskID,  5.5, "Transport control service initialization");
```
Comment (yuriy-larin-firstbridge) [inline apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/app/AplCore.java]: put initialization here
transportInteractionService = CDI.current().select(TransportInteractionService.class).get();

**Pull request merged**